src/pvv/side/door.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace pvv\side;
+
+use \PDO;
+
+class Door{
+	private $pdo;
+
+	public function __construct(PDO $pdo){
+		$this->pdo = $pdo;
+	}
+
+	public function getAll() {
+		$query = 'SELECT time, open FROM door ORDER BY time DESC';
+		$statement = $this->pdo->prepare($query);
+		$statement->execute();
+
+		$doorEvents = [];
+		foreach($statement->fetchAll() as $row){
+			$doorEvents[] = [
+				'time'        => $row['time'],
+				'open'        => $row['open']
+			];
+		}
+
+		return $doorEvents;
+	}
+
+	public function getEntriesAfter($startTime) {
+		$query = 'SELECT time, open FROM door WHERE time > :startTime ORDER BY time DESC';
+		$statement = $this->pdo->prepare($query);
+		$statement->bindParam(':startTime', $startTime, PDO::PARAM_STR);
+		$statement->execute();
+
+		$doorEvents = [];
+		foreach($statement->fetchAll() as $row){
+			$doorEvents[] = [
+				'time'        => (int)$row['time'],
+				'open'        => (bool)$row['open']
+			];
+		}
+
+		return $doorEvents;
+	}
+
+	public function getCurrent() {
+		$query = 'SELECT time, open FROM door ORDER BY time DESC LIMIT 1';
+		$statement = $this->pdo->prepare($query);
+		$statement->execute();
+		$row = $statement->fetch();
+		return [
+			'time'        => (int)$row['time'],
+			'open'        => (bool)$row['open']
+		];
+	}
+
+	private function removeOld() {
```

Comment:
nice, relational databases don't scale well with a lot of timeseries data

how often is it run?

Reply:
The Raspberry Pi(Probaly ESP8266 later) posts the door state once every 15 minutes and immediately if the state changes.
4 times an hour * 24 * 7 = 672, so the database currently stores approximately 700 elements. Can be greatly reduced by only storing actual changes...

Reply:
ah, it's called in createEvent